### PR TITLE
Change to use go module and dependent OTel packages up to date

### LINKED ISSUE #675: Change to use go module and dependent OTel packages up to date
* Remove dep dependency from shippingservice, checkoutservice, productcatalogservice
* Update dependent OTel library version to v0.17.0
* Update dependent proto version to v1.35.0

Replacing #637 and #638 because merging issues.

### PR DESCRIPTION
**Pull request by ymotongpoo**
* Remove dep dependency from shippingservice
* Remove dep dependency from checkoutservice
* Remove dep dependency from productcatalogservice
* Update dependent OTel library version to v0.17.0
* Update dependent proto version to v1.35.0

Fixes #675

### CODE REVIEW

**Comment by ymotongpoo** [comment]
The result of `make deploy` with my own GKE cluster
https://pastebin.com/evTPCRyz

**Comment by ymotongpoo** [comment]
@daniel-sanche do we have the guide to run provisioning test from local environment? Thought the CI is failing on launching `productcatalogservice`, my local test (`make deploy`) worked fine and confirmed all pods up and running. I'm not sure what was the cause of the issue.

Error logs from the CI

```
+ kubectl wait --for=condition=available --timeout=500s deployment/productcatalogservice
error: timed out waiting for the condition on deployments/productcatalogservice
Error: Process completed with exit code 1.
```

**Comment by daniel-sanche** [comment]
@ymotongpoo the best way to test sandbox end-to-end is to launch a full sandbox execution:

If you go to https://cloud-ops-sandbox.dev/, you can right click and copy the address for the open in cloud shell button. That gives you a link like this:

https://console.cloud.google.com/cloudshell/editor?cloudshell_git_repo=https://github.com/GoogleCloudPlatform/cloud-ops-sandbox.git&cloudshell_git_branch=v0.6.0&shellonly=true&cloudshell_image=gcr.io/stackdriver-sandbox-230822/cloudshell-image/uncertified:v0.6.0&cloudshell_tutorial=docs/tutorial.md

You can replace the value for the cloudshell_git_branch (and cloudshell_git_repo if you're using a fork). 

Let me know if that works for you. I can try to take a look in my own environment later this week

File: src/productcatalogservice/Dockerfile [hunk 1]
```diff
@@ -12,29 +12,26 @@
...
 RUN GRPC_HEALTH_PROBE_VERSION=v0.3.5 && \
     wget -qO/bin/grpc_health_probe https://github.com/grpc-ecosystem/grpc-health-probe/releases/download/${GRPC_HEALTH_PROBE_VERSION}/grpc_health_probe-linux-amd64 && \
     chmod +x /bin/grpc_health_probe
 WORKDIR /productcatalogservice
-COPY --from=builder /productcatalogservice ./server
+COPY --from=builder /go/bin/productcatalogservice /productcatalogservice
 COPY products.json .
 EXPOSE 3550
-ENTRYPOINT ["/productcatalogservice/server"]
+ENTRYPOINT ["/productcatalogservice"]
```
Comment (daniel-sanche) [inline src/productcatalogservice/Dockerfile:36]: When I try to deploy this using `make deploy`, I'm seeing this error: 
```
Error: failed to start container "server": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "exec: \"/productcatalogservice\": permission denied": unknown
```

Isn't this a directory? Should it be `/productcatalogservice/productcatalogservice` or `./productcatalogservice` instead? That seemed to fix the issue for me

Comment (ymotongpoo) [reply src/productcatalogservice/Dockerfile:36]: @daniel-sanche ah, yes, you are right. I was too into the Go source code itself and didn't realize the issue here. Thanks for catching

**Comment by ymotongpoo** [comment]
Split this PR to #714 #715 #716. Closing this accordingly.

**Pull request closed without merging**